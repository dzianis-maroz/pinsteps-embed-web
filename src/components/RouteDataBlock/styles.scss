@font-face {
    font-family: "Roboto"; 
    src: url("../../assets/fonts/Roboto.ttf"); 
    font-style: normal; 
    font-weight: normal; 
    };

.route-data-block-style {
    display: flex;
    max-width: 560px;
    min-height: 315px;
    background-color: #F5F4F1;
    border-radius: 16px;
    justify-content: space-evenly;
    font-family: Roboto;
    font-style: normal;
    font-weight: normal;
    position: relative;
};

.left-column {
    width: 204px;
    max-height: 260px;
    margin: 28px 0px;
    position: relative;

    h4 {
        display: inline;
        background: rgba(18, 18, 18, 0.5);
        border-radius: 2px;
        font-size: 12px;
        line-height: 14px;
        color: #FFFFFF;
        padding: 2px 6px;
    };

    p {
        font-size: 18px;
        line-height: 21px;
        color: #121212;
        margin-top: 16px;
    };
};

.route-details {
    display: flex;
    justify-content: space-between;
    margin-bottom: 16px;

    .parameter {
        font-size: 8px;
        line-height: 9px;
        text-transform: uppercase;
        color: #979797;
    };

    .parameter-value {
        font-size: 12px;
        line-height: 14px;
        color: #121212;
        margin: 2px 0px;
    };
};

.author {
    display: flex;
    justify-content: flex-start;
    margin-bottom: 16px;

    p {
        font-size: 12px;
        line-height: 14px;
        color: #8C8C8C;
    };

    .author-photo {
        width: 40px;
        height: 40px;
        margin-right: 16px;
        border-radius: 50%;
        position: relative;
    }

};

.logo {
    position: absolute;
    width: 112px;
    height: 24px;
    left: 7%;
    top: 280px;

    img {
        width: 112px;
        height: 24px;  
    }
};

.right-column {
    width: 215px;
    max-height: 267px;
    margin: 28px 0px;

    img {
        width: 215px;
        height: 215px;
        border-radius: 8px;
        margin-bottom: 16px;
    };

    button {
        width: 215px;
        height: 36px;
        background: #3B48EE;
        border: 1px solid #3B48EE;
        border-radius: 8px;
        font-size: 14px;
        line-height: 16px;
        text-align: center;
        color: #FFFFFF;
        cursor: pointer;
    };

    button:hover {
        border-color: #3B48EE; 
        background: #FFFFFF; 
        color: #3B48EE; 
    };
};

@media (max-width: 530px) {
    .logo {
        left: 6%;
    };
};

@media (max-width: 510px) {
    .logo {
        left: 5%;
    };
};

@media (max-width: 500px) {
    .logo {
        left: 4%;
    };
};

@media (max-width: 490px) {
    .logo {
        left: 3%;
    };
};

@media (max-width: 480px) {
    .route-data-block-style {
        flex-direction: column;
        justify-content: space-between;
        align-items: center;
    };

    .author {
        margin-bottom: 18px;
        justify-content: center;
    };

    .left-column, .right-column {
      margin: 0;  
    };

    .logo {
        position: unset;
        margin: 25px 0px 15px 0px;
    };

    h4 {
        position: relative;
        top: 10px;
    };
};