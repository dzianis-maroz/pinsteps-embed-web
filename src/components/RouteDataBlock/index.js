import PropTypes from 'prop-types';

import { getTimeFromMins, getDistanceFromMeters, createAuthorAvatar} from '../../services/helpers';

import './styles.scss';
import logo from '../../assets/images/pinsteps-logo-black2.png';

export const RouteDataBlock = ({ data }) => {

    const { title, picture, id, person, distance, duration, rating, is_payable, localities, countries } = data;

    const { crop_url } = picture;

    const { facebook_user_id, full_name } = person;

    const routeUrl = `https://pinsteps.page.link/?amv=35&apn=com.pinsteps&efr=0&ibi=com.pinsteps.ios&isi=1271374440&link=https%3A%2F%2Fweb.pinsteps.com%2Fshare%2F${id}`;
    
    return(
        <div id = 'route-data-block' className = 'route-data-block-style'>

            <div className = 'left-column'>
                
                    {Boolean(localities) && localities.length !== 0 ?
                    <h4>{localities[0].name}</h4> 
                    : Boolean(countries) && countries.length !== 0 ?
                    <h4>{countries[0].name}</h4>
                    : null}

                <p>{title}</p>

                <div className = 'route-details'>
                    <div>
                        <p className = 'parameter'>Distance</p>
                        <p className = 'parameter-value'>{getDistanceFromMeters(distance)}</p>
                    </div>
                    <div>
                        <p className = 'parameter'>Time</p>
                        <p className = 'parameter-value'>{getTimeFromMins(duration)}</p>
                    </div>
                    <div>
                        <p className = 'parameter'>Raiting</p>
                        <p className = 'parameter-value'>{rating}</p>
                    </div>
                </div>

                <div className = 'author'>
                
                    <a target='_parent' href = {`https://web.pinsteps.com/profile/${person.id}/tracks`}>

                        {
                            facebook_user_id ? 

                            <img src = {`https://graph.facebook.com/${facebook_user_id}/picture`} 
                            className = 'author-photo'/>

                            : <div className = 'author-photo'>{createAuthorAvatar(full_name)}</div>
                        }
                        
                    </a>

                    <p>{person.full_name}</p>
                </div>

            </div>

            <div className = 'right-column'>
                <img src = {crop_url}/>
                {

                is_payable ? 

                <a target='_parent' href = {routeUrl}><button>Buy route</button></a>   
                   
                : <a target='_parent' href = {routeUrl}><button>Get route</button></a>                
                
                }
            </div>

            <div className = 'logo'>
                <a target='_parent' href = 'https://pinsteps.com?utm_source=embed' target="_blank"><img src = {logo}/></a>  
            </div>
            
        </div>        
    );
};

RouteDataBlock.propTypes = {
    path: PropTypes.string, 
    data: PropTypes.object,
};

export default RouteDataBlock;

