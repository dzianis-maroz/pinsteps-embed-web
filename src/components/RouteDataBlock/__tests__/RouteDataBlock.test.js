import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import RouteDataBlock from "../";

configure({ adapter: new Adapter() });

describe("Should render correctly", () => {
  const props = {
    path: "/some/path",
    data: {
      title: "text",
      picture: { crop_url: "link" },
      id: 220,
      person: { facebook_user_id: "facebook user id", full_name: "full name" },
      distance: 55800,
      duration: 289,
      rating: 2,
      is_payable: false,
      localities: [],
      countries: [],
      milestones: [],
    },
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RouteDataBlock {...props} />);
  })

  it("to Match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("component is defined", () => {
    expect(wrapper).toHaveLength(1);
  });

});


/*   it("should render correctly after props changing", () => {
    const newProps = {
      path: 'N E W',
      data: {
        title: "N E W",
        picture: { crop_url: "N E W" },
        id: 220,
        person: { facebook_user_id: "N E W", full_name: "N E W" },
      }  
    };
    wrapper.setProps({ newProps });
    expect(wrapper.props().data).toEqual(newProps.data);
  }); */