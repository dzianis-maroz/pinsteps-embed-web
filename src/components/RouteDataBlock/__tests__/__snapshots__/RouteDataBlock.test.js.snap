// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<div
  className="route-data-block-style"
  id="route-data-block"
>
  <div
    className="left-column"
  >
    <p>
      text
    </p>
    <div
      className="route-details"
    >
      <div>
        <p
          className="parameter"
        >
          Distance
        </p>
        <p
          className="parameter-value"
        >
          55.8km
        </p>
      </div>
      <div>
        <p
          className="parameter"
        >
          Time
        </p>
        <p
          className="parameter-value"
        >
          4h 49m
        </p>
      </div>
      <div>
        <p
          className="parameter"
        >
          Raiting
        </p>
        <p
          className="parameter-value"
        >
          2
        </p>
      </div>
    </div>
    <div
      className="author"
    >
      <a
        href="https://web.pinsteps.com/profile/undefined/tracks"
      >
        <img
          className="author-photo"
          src="https://graph.facebook.com/facebook user id/picture"
        />
      </a>
      <p>
        full name
      </p>
    </div>
  </div>
  <div
    className="right-column"
  >
    <img
      src="link"
    />
    <a
      href="https://pinsteps.page.link/?amv=35&apn=com.pinsteps&efr=0&ibi=com.pinsteps.ios&isi=1271374440&link=https%3A%2F%2Fweb.pinsteps.com%2Fshare%2F220%3Fl%3Den"
    >
      <button>
        Get route
      </button>
    </a>
  </div>
  <div
    className="logo"
  >
    <a
      href="https://pinsteps.com?utm_source=embed"
      target="_blank"
    >
      <img
        src="pinsteps-logo-black2 1.svg"
      />
    </a>
  </div>
</div>
`;
